### DISCUSSION TITLE
Refinement of hierarchical clustering / heat maps

### DISCUSSION THREAD

**Discussion by hnolCol** [hnolCol/instantclue#46] [Ideas]
Dear all,

thanks to @[szabzola](https://github.com/szabzola) and the issue #42, I am opening a discussion about the enhancement of the appearance of the hierarchical clustering. 

The following enhancements are required:

1. Grouping label and color column labelling overlap (see image below) 
2. Different colours for same category in different columns. 

I propose the following changes to enhance the flexibility: 
New config parameter: 

- [ ] hclust.color.column.label.position (options = ["top","bottom]) - label the color column name x-axis bottom
- [ ] hclust.color.column.show (options (bool) = [True,False])
- [ ] hclust.color.axis.border.left (float in pixel) : distance of color axis on the left to the actual heatmap to increase the space for the grouping label
- [ ] hclust.color.combine.categories (options (bool) = [True, False]) - If set to True, categorical values such as "+" (see image below)  will be combined of different color columns (dragged to the Color Button - in the image: t-test Significant..). If set to False, the same category of different columns will be assigned a different color. 

![image](https://user-images.githubusercontent.com/24191910/154032876-805b2bcd-82f6-4a11-a0e1-ba3508cdd290.png)

**Comment by szabzola** [comment]
Hi,
it would be also useful, if color column elements could be ordered. It accepts integer and numeric columns also, but values are in order of data I think, so colorscale is not according to values. But even in case of categorical (eg. I have some quantiles Q1-Q5) color elements occur in unkown order.
![image](https://user-images.githubusercontent.com/61981001/154239101-528b0891-a529-4382-b83c-b3ccf3127ecc.png)
![image](https://user-images.githubusercontent.com/61981001/154239348-2f71e7cb-db5c-41eb-ab6b-418c6fec02a0.png)

**Reply by hnolCol** [reply]
Couldn't agree more. 
I have added the param:
hclust.color.sort.unique.values (bool) which allows to decide to sort or not. Additionally, another param  bool param (hclust.color.sort.unique.values.ascending) will enable the sorting in ascending/descending order. 

For feature perspective:  it would be indeed good if any color mapping could be resorted in custom order by the user (e.g. what you show above). This will be limited to categorical data.

I am taking care of the numeric data color mapping next, I am aiming for the same look as for scatter plots (e.g. min, max , median) however I think, at least in the new release, the numeric scale (e.g. calculating max and min) will not be adjusted per dragged numeric column but rather be scaled across all values. This might be problematic if you want to show two numeric data with different scales and the user has to take care with appropriate scaling before. But I guess that is fine for the moment.
